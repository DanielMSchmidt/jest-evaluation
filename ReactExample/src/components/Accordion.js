import React, { Component, PropTypes } from 'react';

class Accordion extends Component {
  render() {
    const {
      activeIndex,
      content,
      headlines,
    } = this.props;

    return (
      <ul>
        {headlines.map((headline, index) => (
          <li key={index}>
            <h3>{headline}</h3>
            {index === activeIndex && (
              <p>{content[index]}</p>
            )}
          </li>
        ))}
      </ul>
    );
  }
}

Accordion.propTypes = {
  activeIndex: PropTypes.number.isRequired,
  content: PropTypes.arrayOf(PropTypes.node).isRequired,
  headlines: PropTypes.arrayOf(PropTypes.node).isRequired,
};

export default Accordion;
