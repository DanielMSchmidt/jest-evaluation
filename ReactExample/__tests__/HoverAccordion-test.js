import React from 'react';
import renderer from 'react-test-renderer';

import Accordion from '../src/components/Accordion';

describe('Accordion', () => {
  it('renders the additional content on hover', () => {
    const component = renderer.create(
      <Accordion activeIndex={1} headlines={['Foo', 'Bar', 'Baz']} content={['Foo Content', 'Bar content', 'Baz content']} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
