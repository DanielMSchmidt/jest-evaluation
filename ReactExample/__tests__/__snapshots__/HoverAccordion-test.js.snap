exports[`Accordion renders the additional content on hover 1`] = `
<ul>
  <li>
    <h3>
      Foo
    </h3>
  </li>
  <li>
    <h3>
      Bar
    </h3>
    <p>
      Bar content
    </p>
  </li>
  <li>
    <h3>
      Baz
    </h3>
  </li>
</ul>
`;
